### PR TITLE
Bugfix and add some test cases.

### PR DESCRIPTION
**Pull request by PieerePi**

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -65,3 +65,35 @@ assert(result3.foo.bar === result3.foo)
...
+  [ ],
+}
+
+infos.inbox[0] = infos.tasks[0]
+infos.inbox[1] = infos.tasks[0]
+infos.inbox[2] = infos.tasks[1]
+infos.inbox[3] = infos.tasks[1]
+
+result4 = JSON.stringify(infos, decycle())
+console.log(result4)
```
Comment (YChebotaev) [inline test/index.js]: It should be assertion. Now it's just prints `result4`

File: test/index.js [hunk 2]
```diff
@@ -65,3 +65,35 @@ assert(result3.foo.bar === result3.foo)
...
+
+result4 = JSON.parse(result4, retrocycle())
+console.log(result4)
```
Comment (YChebotaev) [inline test/index.js]: Same here

**Review by YChebotaev — CHANGES_REQUESTED** [review]

File: test/index.js (hunk 1 above)
Comment (PieerePi) [reply test/index.js]: Yeah, I commit again before you comment. :-)

**Comment by YChebotaev** [comment]
@PieerePi Seems ok, merged. Many thanks

**Pull request merged**